perllib/FixMyStreet/App/Controller/Admin/Bodies.pm

```diff
@@ -220,145 +201,181 @@ sub check_for_super_user : Private {
 
-        my $contact = $c->model('DB::Contact')->find_or_new(
-            {
-                body_id => $c->stash->{body_id},
-                category => $category,
-            }
-        );
+    my %errors;
 
-        my $email = $c->get_param('email');
-        $email =~ s/\s+//g;
-        my $send_method = $c->get_param('send_method') || $contact->body->send_method || "";
-        unless ( $send_method eq 'Open311' ) {
-            $errors{email} = _('Please enter a valid email') unless is_valid_email_list($email) || $email eq 'REFUSED';
-        }
+    my $current_category = $c->get_param('current_category') || '';
+    my $current_contact = $c->model('DB::Contact')->find({
+        body_id => $c->stash->{body_id},
+        category => $current_category,
+    });
+    $c->stash->{current_contact} = $current_contact;
 
-        $contact->email( $email );
-        $contact->state( $c->get_param('state') );
-        $contact->non_public( $c->get_param('non_public') ? 1 : 0 );
-        $contact->note( $c->get_param('note') );
-        $contact->whenedited( \'current_timestamp' );
-        $contact->editor( $editor );
-        $contact->endpoint( $c->get_param('endpoint') );
-        $contact->jurisdiction( $c->get_param('jurisdiction') );
-        $contact->api_key( $c->get_param('api_key') );
-        $contact->send_method( $c->get_param('send_method') );
-
-        # Set flags in extra to the appropriate values
-        if ( $c->get_param('photo_required') ) {
-            $contact->set_extra_metadata_if_undefined(  photo_required => 1 );
-        }
-        else {
-            $contact->unset_extra_metadata( 'photo_required' );
-        }
-        if ( $c->get_param('inspection_required') ) {
-            $contact->set_extra_metadata( inspection_required => 1 );
-        }
-        else {
-            $contact->unset_extra_metadata( 'inspection_required' );
-        }
-        if ( $c->get_param('reputation_threshold') ) {
-            $contact->set_extra_metadata( reputation_threshold => int($c->get_param('reputation_threshold')) );
+    my $category = $self->trim( $c->get_param('category') );
+    $errors{category} = _("Please choose a category") unless $category;
+    $errors{note} = _('Please enter a message') unless $c->get_param('note') || FixMyStreet->config('STAGING_SITE');
+
+    my $contact = $c->model('DB::Contact')->find_or_new(
+        {
+            body_id => $c->stash->{body_id},
+            category => $category,
         }
-        if ( my @group = $c->get_param_list('group') ) {
-            @group = grep { $_ } @group;
-            if (scalar @group == 0) {
-                $contact->unset_extra_metadata( 'group' );
-            } else {
-                $contact->set_extra_metadata( group => \@group );
-            }
-        } else {
+    );
+    if ($current_contact && $contact->id && $contact->id != $current_contact->id) {
+        $errors{category} = _('You cannot rename a category to an existing category');
```

Comment:
In the BANES cobrand it's possible to use the same display name for multiple contacts, and to set a contact's display name to the same as another contact's actual name. Should this be disallowed?

Reply:
Seems unlikely that it'll arise, but the former should be easy enough, and I guess the latter theoretically could be a use case (probably not, but shrug)